### PR TITLE
[interfaces/json-rpc] Allow VideoLibrary.Clean by directory

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Add `directory` parameter to `VideoLibrary.Clean`, as done on `VideoLibrary.Scan`

## Motivation and Context

there are two ways to remove items that no longer exist from the Kodi library
call single jsonRPC's (e.g. `VideoLibrary.RemoveEpisode`) for each file or clean the library

The first method (each call per file) is intensive when you have to delete hundreds of files, and have side effect to show loading screen (e.g. if you are doing the operation while watching a video)

The second method you do only one call, at the end of your addon operations, is very fast and easy to do because you do not need a separate job to collect the list of paths to remove.

Why add directory parameter:
- Very fast library cleaning on big video databases
- Avoid removing items from other addons
- Avoid side effects with "non-standard" addons (like Jellyfin causes the removal of the entire library, issue: https://github.com/CastagnaIT/plugin.video.netflix/issues/888)

## How Has This Been Tested?

Tested on windows 10 with Jellyfin library items and netflix library items at same time
When i set VideoLibrary.Clean to netflix folder check only the netflix kodi library path

Another test is done on the issue referenced before, with nf+samba library source

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Application.h [hunk 1]
```diff
@@ -259,8 +259,9 @@ friend class CAppInboundProtocol;
    \brief Starts a video library cleanup.
    \param userInitiated Whether the action was initiated by the user (either via GUI or any other method) or not.  It is meant to hide or show dialogs.
    \param content Content type to clean, blank for everything
+   \param path The path to clean or "" (empty string) for a global clean.
    */
-  void StartVideoCleanup(bool userInitiated = true, const std::string& content = "");
+  void StartVideoCleanup(bool userInitiated = true, const std::string& content = "", const std::string &path = "");
```
Comment (Montellese) [inline xbmc/Application.h:264]: `path` is inconsistent with `strDirectory` in `xbmc/Application.cpp`.

File: xbmc/interfaces/json-rpc/schema/methods.json [hunk 2]
```diff
@@ -1821,7 +1821,8 @@
     "permission": "RemoveData",
     "params": [
       { "name": "showdialogs", "type": "boolean", "default": true, "description": "Whether or not to show the progress bar or any other GUI dialog" },
-      { "name": "content", "type": "string", "default": "video", "enum": [ "video", "movies", "tvshows", "musicvideos" ], "description": "Content type to clean for" }
+      { "name": "content", "type": "string", "default": "video", "enum": [ "video", "movies", "tvshows", "musicvideos" ], "description": "Content type to clean for" },
+      { "name": "directory", "type": "string", "default": "" }
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/schema/methods.json:1825]: Please add a `"description"`.

**Review by Montellese — APPROVED** [review]
Looks good apart from the minors.

File: xbmc/Application.h (hunk 1 above)
Comment (CastagnaIT) [reply xbmc/Application.h]: done

File: xbmc/interfaces/json-rpc/schema/methods.json (hunk 2 above)
Comment (CastagnaIT) [reply xbmc/interfaces/json-rpc/schema/methods.json]: added the description, please check if english syntax is correct

File: xbmc/Application.h [hunk 3]
```diff
@@ -259,8 +259,9 @@ friend class CAppInboundProtocol;
    \brief Starts a video library cleanup.
    \param userInitiated Whether the action was initiated by the user (either via GUI or any other method) or not.  It is meant to hide or show dialogs.
    \param content Content type to clean, blank for everything
+   \param path The path to clean or "" (empty string) for a global clean.
```
Comment (Montellese) [inline xbmc/Application.h:262]: You also need to fix `path` to `strDirectory` here.

File: xbmc/interfaces/json-rpc/schema/methods.json [hunk 4]
```diff
@@ -1821,7 +1821,8 @@
     "permission": "RemoveData",
     "params": [
       { "name": "showdialogs", "type": "boolean", "default": true, "description": "Whether or not to show the progress bar or any other GUI dialog" },
-      { "name": "content", "type": "string", "default": "video", "enum": [ "video", "movies", "tvshows", "musicvideos" ], "description": "Content type to clean for" }
+      { "name": "content", "type": "string", "default": "video", "enum": [ "video", "movies", "tvshows", "musicvideos" ], "description": "Content type to clean for" },
+      { "name": "directory", "type": "string", "default": "", "description": "Path to the directory to cleanup, if not specified, performs global cleanup" }
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/schema/methods.json:1825]: `Path to the directory to clean up; performs a global cleanup if not specified` but I'm no native speaker either.

File: xbmc/Application.h (hunk 3 above)
Comment (CastagnaIT) [reply xbmc/Application.h]: you are right sorry done

File: xbmc/interfaces/json-rpc/schema/methods.json (hunk 4 above)
Comment (CastagnaIT) [reply xbmc/interfaces/json-rpc/schema/methods.json]: good done

**Comment by Montellese** [comment]
Thanks for the changes. jenkins build this please

**Comment by CastagnaIT** [comment]
I have not tested i have no addon ready to this,
but theoretically it should work because the directory sent in the param is searched directly in the kodi database
if someone has the opportunity to confirm

**Pull request merged**